#!/usr/bin/env python

PACKAGE = "wave_sim_config"

from dynamic_reconfigure.parameter_generator_catkin import *

import math

gen = ParameterGenerator()

# Parameters for a single sinusoid wave
gen.add("wave_angle", double_t, 0, "The wave direction [rad]", 0.0, 0.0, 2 * math.pi)
gen.add("wave_period", double_t, 0, "The wave period in [s].", 10.0, 0.0, 30.0)
gen.add("wave_amplitude", double_t, 0, "The wave amplitude [m].", 1.0, 0.0, 10.0)

# Generate the config
exit(gen.generate(PACKAGE, "wave_sim_config", "WaveSinusoid"))
